src/app/core/services/logger.service.ts

```diff
@@ -0,0 +1,159 @@
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class LoggerService {
+  private sessionId = 'sample-session-id'; // Replace with actual session ID logic
+
+  // constructor() {}
+
+  private formatLog(
+    level: string,
+    severity: string,
+    tracebackId: string,
+    screenName: string,
+    source: string,
+    deviceInfo: string,
+    data: string,
+    message: string,
+    stackTrace = '',
+    timestamp = '',
+  ): string {
+    return JSON.stringify({
+      timestamp,
+      level,
+      tracebackId,
+      severity,
+      sessionId: this.sessionId,
+      screenName,
+      source,
+      deviceInfo,
+      data,
+      message,
+      stackTrace,
+    });
+  }
+
+  log(
+    level: string,
+    severity: string,
+    tracebackId: string,
+    screenName: string,
```

Comment:
THis should handle filling up most generic fields like screenName, deviceInfo, source, timestamp etc.
When in use we only add message and stacktrace. Also what is the plan/idea for tracebackId?

Reply:
wrote function for filling these generic fields

The tracebackId will be a unique identifier for each log entry, helping us trace and correlate related logs. It's generated using a combination of the current timestamp and a random value to ensure uniqueness.
(I saw this was added in initial setup for pickleball logger service file, so I added it here too)